src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/SyncMethodInvoker.cs

```diff
@@ -63,17 +63,12 @@ public IAsyncResult InvokeBegin(object instance, object[] inputs, AsyncCallback
 
         public object InvokeEnd(object instance, out object[] outputs, IAsyncResult result)
         {
-            var task = result as Task<Tuple<object, object[]>>;
-            if (task == null)
-            {
-                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentException(SR.SFxInvalidCallbackIAsyncResult));
-            }
-
-            var tuple = task.Result;
+            var tuple = TaskHelpers.ToApmEnd<Tuple<object, object[]>>(result);
             outputs = tuple.Item2;
             return tuple.Item1;
         }
 
+        // This should be changed to ValueTuple instead
```

Comment:
I can remove this comment and try to undo indentation change caused by save if you like

Reply:
Please do remove. Just make the file clean and don't worry about the diff looking clean.

Reply:
comment removed